clippy_lints/src/lifetimes.rs

```diff
@@ -519,6 +522,7 @@ where
             where_predicate_depth: 0,
             bounded_ty_depth: 0,
             generic_args_depth: 0,
+            lifetime_elision_impossible: false,
```

Comment:
I wonder if it would be better to run the checker per function. Then we could use the field in the checker and not the `Usage` struct? But that would be a better refactoring, so I'm happy to leave it as is. Just wanted to throw the idea out there.

Reply:
I don't know if it could be done easily, as it also considers the lifetime from enclosing items.